pkg/i2gw/providers/ingressnginx/server_alias.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingressnginx
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/intermediate"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/notifications"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// serverAliasFeature processes the server-alias annotation for ingress-nginx.
+// https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#server-alias
+func serverAliasFeature(ingresses []networkingv1.Ingress, _ map[types.NamespacedName]map[string]int32, ir *intermediate.IR) field.ErrorList {
+	var errs field.ErrorList
+
+	// Keep track of server aliases per gateway
+	serverAliasesByGateway := make(map[types.NamespacedName][]string)
+
+	for _, ingress := range ingresses {
+		if serverAlias := ingress.Annotations["nginx.ingress.kubernetes.io/server-alias"]; serverAlias != "" {
```

Comment:
a nit, but a bit important IMO:

```suggestion
		if serverAlias, ok := ingress.Annotations["nginx.ingress.kubernetes.io/server-alias"]; ok && serverAlias != "" {
```

The reason is so you check first if the key on the map exists, and then you check its value